### PR TITLE
home-assistant: add sonos component dependencies

### PR DESCRIPTION
**Pull request by oneingan**
###### Motivation for this change
Add required python packages and binding them in the component-packages.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
cc @dotlambda

**Comment by FRidh** [comment]
@GrahamcOfBorg build python2.pkgs.psonos python3.pkgs.sonos

File: pkgs/development/python-modules/pysonos/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  propagatedBuildInputs = [ xmltodict requests ifaddr ];
+
+  # Need to override pytest-conv to an old version
+  doCheck = false;
+
+  meta = {
+    homepage = https://github.com/amelchio/pysonos;
+    description = "A SoCo fork with fixes for Home Assistant.";
+    license = lib.licenses.mit;
+  };
```
Comment (dotlambda) [inline pkgs/development/python-modules/pysonos/default.nix]: `maintainers` is missing

File: pkgs/development/python-modules/pysonos/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1pfm23v2mj8l878ilpy9xb0jbkpncnwyf4432haraswvf1cg2bxh";
+  };
+
+  propagatedBuildInputs = [ xmltodict requests ifaddr ];
+
+  # Need to override pytest-conv to an old version
+  doCheck = false;
```
Comment (dotlambda) [inline pkgs/development/python-modules/pysonos/default.nix]: You mean pytest-cov? Just remove the version constraint using `substituteInPlace` in `postPatch`.

**Review by dotlambda — CHANGES_REQUESTED** [review]
It does not make much sense to merge this before #60217 is merged. So I'll ask you to rebase once that's done.

File: pkgs/development/python-modules/pysonos/default.nix (hunk 2 above)
Comment (juaningan) [reply pkgs/development/python-modules/pysonos/default.nix]: of course,
`` pytest-cov>=2.4.0,<2.6 ```
in ``requirements-dev.txt`` file. Ok, I will try with 2.6.1 then.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please update pysonos to 0.0.13 and re-run `parse-requirements.py` after rebasing on d280603.

**Comment by juaningan** [comment]
Add maintainers, update version to 0.0.13 and enable python checks.

File: pkgs/development/python-modules/pysonos/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/pysonos/default.nix]: You need to add yourself to `maintainers/maintainer-list.nix`.

File: pkgs/development/python-modules/pysonos/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  propagatedBuildInputs = [ xmltodict requests ifaddr ];
+
+  checkInputs = [
+    pytest_3 pylint flake8 graphviz
+    mock sphinx sphinx_rtd_theme
+  ];
+
+  meta = {
+    homepage = https://github.com/amelchio/pysonos;
+    description = "A SoCo fork with fixes for Home Assistant.";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pysonos/default.nix]: no period: https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please squash your commits such that there's one called `pythonPackages.pysonos: init at 0.0.13` and another one called `home-assistant: run parse-requirements.py`.

File: pkgs/development/python-modules/pysonos/default.nix (hunk 1 above)
Comment (juaningan) [reply pkgs/development/python-modules/pysonos/default.nix]: done

File: pkgs/development/python-modules/pysonos/default.nix (hunk 3 above)
Comment (juaningan) [reply pkgs/development/python-modules/pysonos/default.nix]: done

File: pkgs/development/python-modules/pysonos/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  checkInputs = [
+    pytest_3 pylint flake8 graphviz
+    mock sphinx sphinx_rtd_theme
+  ];
+
+  meta = {
+    homepage = https://github.com/amelchio/pysonos;
+    description = "A SoCo fork with fixes for Home Assistant";
+    license = lib.licenses.mit;
+    maintainers = "juaningan";
```
Comment (dotlambda) [inline pkgs/development/python-modules/pysonos/default.nix]:
```suggestion
    maintainers = with lib.maintainers; [ juaningan ];
```

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.pysonos python3.pkgs.pysonos

**Comment by dotlambda** [comment]
Sorry, I f**d up. Let's continue here: https://github.com/NixOS/nixpkgs/pull/61748

**Pull request closed without merging**